docs/DeveloperGuide.md

```diff
@@ -7,26 +7,42 @@ title: Developer Guide
 
 --------------------------------------------------------------------------------------------------------------------
 
-## **Acknowledgements**
+## **Acknowledgements** (OUTDATED)
 
 * {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
 --------------------------------------------------------------------------------------------------------------------
 
-## **Setting up, getting started**
+## **Setting up, getting started** (OUTDATED)
 
 Refer to the guide [_Setting up and getting started_](SettingUp.md).
 
 --------------------------------------------------------------------------------------------------------------------
 
 ## **Design**
 
+RC4HDB has the idea of providing a flexible product which functions well, yet not too complex. Keeping this in mind, 
+the developers are pursuing an iterative approach, adding new features and functionalities amidst the evolving requirements.
+This gives rise to the following main guiding principles for RC4HDB: 
+
+**Maintainability**
+
+This project was adapted from an application called [`AddressBook Level 3 (AB3)`](https://se-education.org/addressbook-level3/).
+The developers of `AB 3` have developed in a manner that facilitates easy modification of components. This design allows the functionalities
```

Comment:
This part sounds weird to me, there is an excessive use of **develop**. Perhaps we could change it to "**`AB 3` was developed in a manner that facilitates easy modification of components.**"

Reply:
Should we not stick to `AB3` instead of `AB 3`? (this is following the acronym from line 30)